### PR TITLE
Add org.freedesktop.Sdk.Extension.mingw-w64

### PR DESCRIPTION
**Pull request by gasinvein**
MinGW-w64, toolchain for building applications for Windows.
Primarily this is needed for Wine-related apps, since Wine started to build it's modules as PE binaries. Also someone may find it useful for development with IDEs.
I've tested it only with Wine and DXVK/D9VK, so people more familiar with mingw are welcome to review it, it's possible that I did some mistakes.

### CODE REVIEW

**Comment by LeandroStanger** [comment]
bot, build org.freedesktop.Sdk.Extension.mingw-w64

**Comment by gasinvein** [comment]
For some reason rofiles-fuse prevents flatpak-builder from stripping binaries:
```
make[1]: Nothing to be done for 'install-target'.
make[1]: Leaving directory '/run/build/binutils-mingw-w64/_flatpak_build'
stripping: files/bin/x86_64-w64-mingw32-size
FB: Running: strip --remove-section=.comment --remove-section=.note --strip-unneeded /home/gasinvein/.cache/flatpak-builder/state/rofiles/rofiles-RxCkKZ/files/bin/x86_64-w64-mingw32-size
stripping: files/bin/x86_64-w64-mingw32-objdump
FB: Running: strip --remove-section=.comment --remove-section=.note --strip-unneeded /home/gasinvein/.cache/flatpak-builder/state/rofiles/rofiles-RxCkKZ/files/bin/x86_64-w64-mingw32-objdump
strip: unable to copy file '/home/gasinvein/.cache/flatpak-builder/state/rofiles/rofiles-RxCkKZ/files/bin/x86_64-w64-mingw32-objdump'; reason: Read-only file system
Error: module binutils-mingw-w64: Child process exited with code 1
FB: unmounting rofiles-fuse /home/gasinvein/.cache/flatpak-builder/state/rofiles/rofiles-RxCkKZ
```

bot, build org.freedesktop.Sdk.Extension.mingw-w64

File: org.freedesktop.Sdk.Extension.mingw-w64.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="runtime">
+  <id>org.freedesktop.Sdk.Extension.mingw-w64</id>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-3.0 and LGPL-2.0-or-later and ZPL-2.1 and Public Domain</project_license>
```
Comment (gasinvein) [inline org.freedesktop.Sdk.Extension.mingw-w64.metainfo.xml]: appstream-util doesn't like the `Public Domain` licence. I'm not sure what to put here - MinGW toolchain consists of multiple components under different licences.

**Comment by gasinvein** [comment]
bot, build org.freedesktop.Sdk.Extension.mingw-w64

**Comment by LeandroStanger** [comment]
bot, build org.freedesktop.Sdk.Extension.mingw-w64

**Pull request closed without merging**